src/Components/Color/Color.jsx

```diff
@@ -47,23 +39,23 @@ export default function Color({ color, onDelete }) {
               <button type="button"  onClick={plsCancel}>
                 Cancel
               </button>
-              <button type="button" onClick={() => onDelete(currentColor.id)}>Yes Delete</button>
+              <button type="button" onClick={() => onDelete(color.id)}>Yes Delete</button>
             </section>
           {/* -----Hidden Buttons -------------------------------------------------------------------- */}
 
       <button type="button" className={isEdit ? "" : "hidden"} onClick={plsEdit}>Edit </button>
           {/* -----Hidden edit -------------------------------------------------------------------- */}
           <section className={isEdit ? "hidden" : ""}>
-            <form onSubmit={handleEdit} >
-              <label htmlFor="role">Role:</label>{/* <br/> */}
-              <input type="text" id="role" name="role" defaultValue={currentColor.role} />
+            <form onSubmit={(event) => onHandleEdit(color.id, event)} >
+              <label htmlFor="role">Role:</label>
+              <input type="text" id="role" name="role" defaultValue={color.role} />
               <label htmlFor="hex">Hex:</label>
               <div>
-                  <Colorinput id="hex" defaultValue={currentColor.hex}/>
+                  <Colorinput id="hex" defaultValue={color.hex}/>
               </div>
               <label htmlFor="contrastText">Contrast text:</label>
               <div>
-                  <Colorinput id="contrastText" defaultValue={currentColor.contrastText} />
+                  <Colorinput id="contrastText" defaultValue={color.contrastText} />
               </div>
               <input type="submit" className="inputBtn" value="Update Color" />
             </form>
```

Comment:
You should create a `<ColorForm />` component. Because you have to use it at different places.